purchase_last_price_info/models/product.py

```diff
@@ -14,16 +11,22 @@ def _get_last_purchase(self):
         """ Get last purchase price, last purchase date and last supplier """
         lines = self.env['purchase.order.line'].search(
             [('product_id', '=', self.id),
-             ('state', 'in', ['confirmed', 'done'])]).sorted(
+             ('state', 'in', ['purchase', 'done'])]).sorted(
             key=lambda l: l.order_id.date_order, reverse=True)
         self.last_purchase_date = lines[:1].order_id.date_order
         self.last_purchase_price = lines[:1].price_unit
         self.last_supplier_id = lines[:1].order_id.partner_id
 
     last_purchase_price = fields.Float(
-        string='Last Purchase Price', compute='_get_last_purchase')
+        string='Last Purchase Price',
+        compute='_get_last_purchase',
+    )
```

Comment:
Add the `digits=dp.get_precision('Product Price')` for display the same decimal precisions as in the lines.